data/tutorials/language/3mt_04_metaprogramming.md

```diff
@@ -21,21 +21,20 @@ preprocessor](https://github.com/ocaml-ppx/ppxlib/tree/main/examples/simple-exte
 
 <!-- $MDX skip -->
 ```ocaml
-Printf.printf "This program has been compiled on %s" [%get_env "OSTYPE"]
+Printf.printf "This program has been compiled by user: %s" [%get_env "USER"]
+
 ```
 
-would become:
+When you compile the code with the preprocessor, it will replace [%get_env "USER"] with the content of the USER environment variable. If the USER environment variable is set to "JohnDoe" for example, the line would become:
 
 <!-- $MDX skip -->
 ```ocaml
-Printf.printf "This program has been compiled on %s" "linux-gnu"
+Printf.printf "This program has been compiled on %s" "JohnDoe"
```

Comment:
The text here doesn't match the text in line 24.

Reply:
it is because when we compile the code with the preprocessor , it will replace "USER" with the content of the "USER" environment variable .Now , if the USER environment variable is set to "JohnDoe" then in that case it will print "JohnDoe". I have explained all this in the commit above .Kindly review @sabine @SaySayo

Reply:
It is because when we compile the code with the preprocessor , it will replace "USER" with the content of the "USER" environment variable .Now , if the USER environment variable is set to "JohnDoe" then in that case it will print "JohnDoe". I have explained all this in the commit above .Kindly review @sabine @SaySayo

Reply:
I estimate that a new user would potentially be confused when they read that

```
Printf.printf "This program has been compiled by user: %s" [%get_env "USER"]
```

is transformed to

```
Printf.printf "This program has been compiled on %s" "JohnDoe"
```

by the preprocessor.

---

`compiled by user:` vs `compiled on`

Reply:
I have made the required change in my most recent commit.Kindly review @sabine